src/http.cc

```diff
@@ -1441,9 +1441,13 @@ HttpStateData::writeReplyBody()
     else if (clen < 0 && eof)
         parsedWhole = "http parsed body ending with expected/required EOF";
 
-    if (parsedWhole)
+    if (parsedWhole) {
         markParsedVirginReplyAsWhole(parsedWhole);
-    else if (eof)
+#if USE_ADAPTATION
+        if (!startedAdaptation)
```

Comment:
We cannot use doneWithAdaptation() here because it is may return true when adaptation was started and failed. writeReplyBody() and decodeAndWriteReplyBody() are called after startAdaptation(), so startedAdaptation must have been initialized with a correct value here. Successful adaptation outcomes are covered by endAdaptedBodyConsumption().

Reply:
Done.